MaskRCNN_no_batch_convergence/model_box.py

```diff
@@ -9,13 +9,38 @@
 
 from config import config
 
+@under_name_scope()
+def clip_boxes_batch(boxes, window, batch_ids, name=None):
+    """
+    Args:
+        boxes: nx(#class)x4, xyxy
+        window: BSx2 
+        batch_ids: 1-D Tensor of size n
+    """
+    boxes = tf.maximum(boxes, 0.0)
+
+    wh = tf.reverse(tf.gather(window, batch_ids), axis=[1]) 
+    whwh = tf.concat((wh, wh), axis=1)
+
+    multiples = tf.stack([tf.constant(1), tf.shape(boxes)[1], tf.constant(1)])
+
+    whwh_tiled = tf.tile(tf.expand_dims(whwh, 1), multiples)    # asd = [1, #class, 1]
+
+#    m = tf.tile(tf.reverse(window, [0]), [2])    # (4,)
+    boxes = tf.minimum(boxes, tf.cast(whwh_tiled, tf.float32), name=name)
+    return boxes
+
+
+
 @under_name_scope()
 def clip_boxes(boxes, window, name=None):
     """
     Args:
         boxes: nx4, xyxy
         window: [h, w]
     """
+    window = tf.squeeze(window, axis=0)
```

Comment:
@karakusc Is this an issue for batch-by-default?

Reply:
I wrote `clip_boxes_batch` above this, we should be using that one.